llvm/lib/SYCLLowerIR/SYCLPropagateAspectsUsage.cpp

```diff
@@ -225,6 +225,85 @@ const AspectsSetTy &getAspectsFromType(const Type *T,
   return Result;
 }
 
+/// Utility function to identify FP64 conversion instruction
+bool isFP64ConversionInstruction(const Instruction &I) {
+  switch (I.getOpcode()) {
+  default:
+    return false;
+  case Instruction::Alloca:
+  case Instruction::GetElementPtr:
+  case Instruction::Load:
+  case Instruction::Store:
+  case Instruction::FPExt:
+  case Instruction::FPTrunc:
+  case Instruction::SIToFP:
+  case Instruction::UIToFP:
+  case Instruction::FPToUI:
+  case Instruction::FPToSI:
+  case Instruction::FCmp:
+  case Instruction::PHI:
+  case Instruction::ExtractValue:
+  case Instruction::InsertValue:
+  case Instruction::Ret:
+    return true;
+  // In case of call instructions, we check if the definition of the called
+  // function is present inside the current module. If present, we conclude
+  // that the call instruction does not require FP64 computations and return
```

Comment:
I wonder if we could handle this case if we run the pass in sycl-post-link where the module is complete. If we don't expect this will be a likely thing users can hit, we could just document this.

Reply:
Sorry, I didn't mean to move the entire pass, my idea is basically the same as yours.

Reply:
I added a TODO for this.

Thanks